cmd/crank/pkg/lint.go

```diff
@@ -0,0 +1,129 @@
+/*
+Copyright 2020 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pkg
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+
+	"github.com/crossplane/crossplane-runtime/pkg/parser"
+	"github.com/crossplane/crossplane/pkg/controller/pkg/revision"
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/client"
+	"github.com/docker/docker/pkg/stdcopy"
+)
+
+type LintCmd struct {
+	Image  string          `arg:"" optional:"" name:"image" help:"Name of the package image to run and lint the output of."`
+	NoPull bool            `help:"Flag to disable pulling the base image. This is useful if you want to build from a local copy of a base image that does not exist or resides in an inaccessible registry."`
+	ctx    context.Context // sneaky context in case this is invoked by another cmd
+	docker *client.Client  // sneaky docker client in case this is invoked by another cmd that already made one
+}
+
+func (c *LintCmd) Run() error {
+	if c.ctx == nil {
+		c.ctx = context.Background()
+	}
+	if c.docker == nil {
+		var err error
+		c.docker, err = client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
+		if err != nil {
+			return err
+		}
+	}
+
+	if !c.NoPull {
+		dra, err := NewDockerRegistryAuth(c.ctx, c.Image)
+		options, err := dra.BuildPullOpts(c.ctx)
+		if err != nil {
+			return err
+		}
+		resp, err := c.docker.ImagePull(c.ctx, dra.NormalizedImageName(), options)
+		if err != nil {
+			return err
+		}
+		_, err = io.Copy(os.Stdout, resp)
+		if err != nil {
+			return err
+		}
+		err = resp.Close()
+		if err != nil {
+			return err
+		}
+	}
+
+	outBuf, _, err := c.RunContainerGetLogs()
+	if err != nil {
+		return err
+	}
+
+	metaScheme, _ := revision.BuildMetaScheme()
+	objScheme, _ := revision.BuildObjectScheme()
+	p := parser.New(metaScheme, objScheme)
+	pkg, err := p.Parse(c.ctx, outBuf)
```

Comment:
You will need to check this error as it later gets overwritten by linter output

Reply:
ironically the go linter would have saved me from myself here 😭

Reply:
fixed